public/app/plugins/datasource/loki/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -138,6 +149,24 @@ const MonacoQueryField = ({ history, onBlur, onRunQuery, initialValue, datasourc
     editor.onDidChangeModelContent(checkDecorators);
   };
 
+  let typingTimer: NodeJS.Timeout;
+
+  const onType = (query: string) => {
+    clearTimeout(typingTimer);
+
+    typingTimer = setTimeout(() => {
+      if (isValidQuery(query) === false) {
+        return;
+      }
+
+      const timerange = datasource.getTimeRange();
+      const previousTimerange = undefined;
+      const previousQuery = undefined;
+
+      makeStatsRequest(datasource, timerange, previousTimerange, query, previousQuery, setQueryStats!);
```

Comment:
I am wondering if we should make this a more "generic" action, not coupled to "stats", but something like an action/callback that is called whenever the query changed. WDYT? Also WDYT @ivanahuckova?

Reply:
I don't mind having this coupled with stats at the moment. I personally would wait until we have something else that should be run, but I can see your point as well. I don't have strong opinion here.